lib/portfinder.js

```diff
@@ -76,40 +76,52 @@ exports.getPort = function (options, callback) {
     options = {};
   }
 
-  options.host = options.host || '::1';
-  var hosts = [options.host, '0.0.0.0', '127.0.0.1'];
+  var hosts = ['::1', '127.0.0.1', '0.0.0.0'];
+  if (options.host) {
+    hosts.push(options.host);
+  }
 
   // remove duplicate host arg if already in hosts array
-  for (var i = 1; i < hosts.length; i++) {
-    if (options.host === hosts[i]) {
-      hosts.shift();
-      break;
-    }
-  }
+  var uniqueHosts = hosts.filter(function(host, index) {
+    return hosts.indexOf(host) == index;
+  });
 
-  // create array of eventual found ports for each host
-  var testPortFunctions = [];
-  for (var j = 0; j < hosts.length; j++) {
-    (function() {
-      var jj = j;
-      var testPortFn = function(callback) {
-        internals.testPort({ host: hosts[jj], port: options.port }, callback);
-      };
-
-      testPortFunctions.push(testPortFn);
-    }());
+  function findCommonOpenPort(portToTest) {
+    var ports = [];
+    var error;
+    async.everyLimit(uniqueHosts, 1, function(host, next) {
```

Comment:
so it does seem that asyncLimit does iterate through *until a falsy value is returned* - this would be fine except we need to handle an obscure error pointed out in https://github.com/indexzero/node-portfinder/issues/26

Basically not every user of every OS or corporate issued laptop, cloud booted server, etc - will have ipv6 open - so we need to catch the error that occurs when that interface is not available. That error is `EADDRNOTAVAIL` - there is always a chance I'm not fully correct about why or when this error will be thrown, but it will throw for some users - so we need to catch it and remove that host from the array when it is thrown, continue testing the rest of the hosts, and never test it again as it will always throw(i believe) for any port we try to lookup for it

aka, we catch this error, remove that host from the array, continue, never test that host again

this seems to affect the way this iterator is written in a couple of ways